crates/burn-import/src/onnx/coalesce.rs

```diff
@@ -50,21 +50,23 @@ pub(crate) fn convert_gemm_to_linear(node: &mut Node) {
         node.attrs.remove("transB");
 
         // Transpose the weights
-        transpose_linear_node_weights(node);
+        transpose_linear_node_weights(node, graph_io);
     } else {
         panic!("Full Gemm node not supported yet.");
     }
 }
 
 // Transpose linear weights (required for Gemm -> Linear conversion)
-fn transpose_linear_node_weights(node: &mut Node) {
+fn transpose_linear_node_weights(node: &mut Node, graph_data: &GraphData) {
     assert!(
         node.inputs.len() > 1,
         "Linear node must have at least 2 input"
     );
 
     assert!(node.inputs[1].value.is_some(), "Input must have a value");
 
+    assert!(graph_data.initializers.contains_key(&node.inputs[1].name), "Currently only initializers are supported for weights.\nPlease open an issue if you encounter this error.");
+
```

Comment:
it's related to what I mentioned on the discord channel. The spec seems to support passing the weights from another node, and the current function modifies the inputs (transposing the weights). I suspect this will lead to a compilation error due to a shape mismatch, since the transpose never happened in the generated code

If that is the case, the actual fix would be to add a new transpose node before the current one that produces that output, but I'm not sure anyone would encounter this in the wild. 

I just realized this will incorrectly fail if the weight was both an input and initializer. 

I can fix that or remove the check, whichever you prefer

Reply:
I removed the check for now. It can be added later if there's a need for it